cryptoki/src/session/object_management.rs

```diff
@@ -96,6 +96,41 @@ impl Session {
         }
     }
 
+    /// Copy an object
+    /// A optional template can be provided to change some attributes of the new object, when allowed.
+    ///
+    /// # Arguments
+    ///
+    /// * `object` - The [ObjectHandle] used to reference the object to copy
+    /// * `template` - an optional reference to a slice of attributes, in the form of `Some(&[Attribute])`, 
+    ///    or set to `None`` if not needed.
+    ///
+    /// # Returns
+    ///
+    /// This function will return a new [ObjectHandle] that references the newly created object.
+    ///
+    pub fn copy_object(&self, object: ObjectHandle, template: Option<&[Attribute]>) -> Result<ObjectHandle> {
+        let mut template: Vec<CK_ATTRIBUTE> = match template {
+            Some(template) => template.iter().map(|attr| attr.into()).collect(),
+            None => Vec::new(),
+        };
```

Comment:
Maybe you can do what we do in `find_objects` and not use the `Option`:
```rust
let mut template: Vec<CK_ATTRIBUTE> = template.iter().map(|attr| attr.into()).collect();
```

if it works :detective: !

Reply:
I found using Option<> to be more idiomatic. OTOH there is maybe little reason for not providing a template, when doing a copy. I'll adapt the PR if this is preferred.

Reply:
Sure thing. Let me adapt the code.